addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -4125,7 +4127,25 @@ export class OdooEditor extends EventTarget {
             this.execCommand('strikeThrough');
         } else if (IS_KEYBOARD_EVENT_LEFT_ARROW(ev) || IS_KEYBOARD_EVENT_RIGHT_ARROW(ev)) {
             const side = ev.key === 'ArrowLeft' ? 'previous' : 'next';
-            const { anchorNode, anchorOffset } = this.document.getSelection() || {};
+            const selection = this.document.getSelection();
+            const { anchorNode, anchorOffset, focusNode, focusOffset } = selection || {};
+            if (ev.shiftKey) {
+                const currentBlock = closestBlock(focusNode);
+                const isSelectionAtStart = firstLeaf(currentBlock) === focusNode && focusOffset === 0;
+                const isSelectionAtEnd = lastLeaf(currentBlock) === focusNode && focusOffset === nodeSize(focusNode);
+                const adjacentBlock = side === 'previous' ? currentBlock.previousElementSibling : currentBlock.nextElementSibling;
+                const targetBlock = side === 'previous' ? adjacentBlock?.previousElementSibling : adjacentBlock?.nextElementSibling;
+                if (targetBlock && !adjacentBlock.isContentEditable) {
```

Comment:
That's nitpicking but I would test for `!adjacentBlock.isContentEditable` before checking for `targetBlock` because it's the whole reason why this block of code exists. If `adjacentBlock` is contentEditable then none of this matters. In particular, I think the whole block of code needs a comment because the reader needs to read through quite a few lines before they can get a sense of what we're doing an why we're doing it, that is handling keyboard selection around non-editables.

Reply:
> I think the whole block of code needs a comment

How about:
```js
// Since selection can't traverse contenteditable="false"
// elements, we adjust the selection to the sibling of
// contenteditable="false" element.
```

Reply:
Perfect, thanks! 👍